src/main/java/org/gradle/profiler/jprofiler/JProfilerController.java

```diff
@@ -27,7 +27,14 @@ public JProfilerController(ScenarioSettings settings, JProfilerConfig jProfilerC
     }
 
     @Override
-    public void start() throws IOException, InterruptedException {
+    public void startSession() throws IOException, InterruptedException {
+        if (!profileWholeLifeTime()) {
```

Comment:
this feels like something that could perhaps be taken care of at some higher level, as it's not really honouring the contract. The main loop might be aware in some way of profiling that is applied to the whole process lifetime and can avoid switching it on and off, possibly by simply using a no-op controller when using whole process profiling.

Reply:
It's now using a no-op control in daemon-mode, just like the YourKitProfiler already did.